solidity/src/BftBridge.sol

```diff
@@ -187,16 +196,88 @@ contract BFTBridge is TokenManager, UUPSUpgradeable, OwnableUpgradeable, Pausabl
         controllerAccessList[controller] = false;
     }
 
-    /// Main function to withdraw funds
+    /// Transfer funds to user according the signed encoded order.
     function mint(
         bytes calldata encodedOrder
     ) external whenNotPaused {
-        uint256 initGasLeft = gasleft();
+        MintOrderData memory order = _decodeOrder(encodedOrder[:MINT_ORDER_DATA_LEN]);
+        _validateOrder(order);
+        _checkMintOrderSignature(encodedOrder);
 
-        MintOrderData memory order = _decodeAndValidateOrder(encodedOrder[:269]);
+        uint256 feeAmount = 0;
+        if (_isFeeRequired()) {
+            feeAmount = (COMMON_BATCH_MINT_GAS_FEE + ORDER_BATCH_MINT_GAS_FEE) * tx.gasprice;
+        }
 
-        _checkMintOrderSignature(encodedOrder);
+        _mintInner(order, feeAmount);
+
+    }
 
+    /// Transfer funds to users according the signed encoded orders.
+    /// Returns `processedOrders` boolean array: 
+    /// `processedOrders[i] == true`, means `encodedOrders[i]` successfully processed.
+    function batchMint(
```

Comment:
This function really asks for some unit tests....

Reply:
I had the same thoughts)
https://infinityswap.atlassian.net/browse/EPROD-1001

In this PR, most common workflow will be tested through the integration test. Test for other flows will be implemented separetely.